tests/Api/Parser/JsonParserTest.php

```diff
@@ -0,0 +1,213 @@
+        $client = $this->generateTestClient($service, ['timestamp' => []]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $this->setExpectedException(ParserException::class);
+        $handler($command)->wait()['timestamp']->__toString();
+    }
+    public function testEpochArrayReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => [932169600]]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $this->setExpectedException(ParserException::class);
+        $handler($command)->wait()['timestamp']->__toString();
+    }
+
+    public function testMixedStringReturnedFromService()
+    {
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => "932169600abc"]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $this->setExpectedException(ParserException::class);
+        $handler($command)->wait()['timestamp']->__toString();
+    }
+
+    public function testDateTimeReturnedFromService()
+    {
+        $datetime = new DateTime();
+        $service = $this->generateTestService();
+        $client = $this->generateTestClient($service, ['timestamp' => $datetime]);
+        $command = $client->getCommand('ParseJson');
+        $list = $client->getHandlerList();
+        $handler = $list->resolve();
+        $this->setExpectedException(ParserException::class);
+        $handler($command)->wait()['timestamp']->__toString();
+    }
+
+    private function generateTestClient(Service $service, $args = [])
+    {
+        return new AwsClient(
+            array_merge(
+                [
+                    'service'      => 'foo',
+                    'api_provider' => function () use ($service) {
+                        return $service->toArray();
+                    },
+                    'region'       => 'us-east-1',
+                    'version'      => 'latest',
+                    'http_handler' => function () use ($args) {
+                        return new FulfilledPromise(new Response(200, [], json_encode($args)));
+                    }
+                ],
+                $args
+            )
+        );
+    }
+
+    private function generateTestService()
+    {
+        return new Service(
+            [
+                'metadata' => [
+                    "protocol" => "json",
+                    "apiVersion" => "2014-01-01"
+                ],
+                'shapes' => [
+                    "ParseJsonRequest" => [
+                        "type" => "structure",
+                        "members" => [
+                            "timestamp" =>[
```

Comment:
This is a super nitpick that doesn't really matter in a test model, but shapes and members are supposed to be capitalized (a few services deviate, but those are legacy models).